pkg/scalers/azure_eventhub.go

```diff
@@ -0,0 +1,127 @@
+	if err != nil {
+		return azure.Environment{}, &azblob.SharedKeyCredential{}, fmt.Errorf("unable to parse connection string: %s", storageConnection)
+	}
+
+	azureEnv, err := azure.EnvironmentFromName(environmentName)
+	if err != nil {
+		return azureEnv, nil, fmt.Errorf("could not get azure.Environment struct: %s", err)
+	}
+
+	cred, err := azblob.NewSharedKeyCredential(storageAccountName, storageAccountKey)
+	if err != nil {
+		return azureEnv, nil, fmt.Errorf("could not prepare a blob storage credential: %s", err)
+	}
+
+	return azureEnv, cred, nil
+}
+
+// GetEventHubClient returns eventhub client
+func GetEventHubClient(connectionString string) (*eventhub.Hub, error) {
+	hub, err := eventhub.NewHubFromConnectionString(connectionString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create hub client: %s", err)
+	}
+
+	return hub, nil
+}
+
+// GetCheckpointFromBlobStorage accesses Blob storage and gets checkpoint information of a partition
+func GetCheckpointFromBlobStorage(ctx context.Context, partitionID string, eventHubMetadata EventHubMetadata) (Checkpoint, error) {
+	storageAccountName, _, err := ParseAzureStorageConnectionString(eventHubMetadata.storageConnection)
+	if err != nil {
+		return Checkpoint{}, fmt.Errorf("unable to parse storage connection string: %s", err)
+	}
+
+	eventHubNamespace, eventHubName, err := ParseAzureEventHubConnectionString(eventHubMetadata.eventHubConnection)
+	if err != nil {
+		return Checkpoint{}, fmt.Errorf("unable to parse event hub connection string: %s", err)
+	}
+
+	// TODO: add more ways to read from different types of storage and read checkpoints/leases written in different JSON formats
+	u, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/azure-webjobs-eventhub/%s/%s/%s/%s", storageAccountName, eventHubNamespace, eventHubName, eventHubMetadata.eventHubConsumerGroup, partitionID))
+
+	_, cred, err := GetStorageCredentials(eventHubMetadata.storageConnection)
+	if err != nil {
+		return Checkpoint{}, fmt.Errorf("unable to get storage credentials: %s", err)
+	}
+
+	// Create a BlockBlobURL object to a blob in the container.
+	blobURL := azblob.NewBlockBlobURL(*u, azblob.NewPipeline(cred, azblob.PipelineOptions{}))
+
+	get, err := blobURL.Download(ctx, 0, 0, azblob.BlobAccessConditions{}, false)
+	if err != nil {
+		return Checkpoint{}, fmt.Errorf("unable to download file from blob storage: %s", err)
+	}
+
+	blobData := &bytes.Buffer{}
+	reader := get.Body(azblob.RetryReaderOptions{})
+	blobData.ReadFrom(reader)
+	defer reader.Close() // The client must close the response body when finished with it
+
+	var dat Checkpoint
+
+	if err := json.Unmarshal(blobData.Bytes(), &dat); err != nil {
+		return Checkpoint{}, fmt.Errorf("failed to decode blob data: %s", err)
+	}
+
+	return dat, nil
+}
+
+func ParseAzureEventHubConnectionString(connectionString string) (string, string, error) {
+	parts := strings.Split(connectionString, ";")
+
+	var eventHubNamespace, eventHubName string
```

Comment:
Since you're doing string parsing in this function, it's good practice to add a comment with an example connection string that you expect to parse.